testing/sdk_login.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import os
+
+import requests
+import shakedown
+
+__CLI_LOGIN_OPEN_TOKEN = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik9UQkVOakZFTWtWQ09VRTRPRVpGTlRNMFJrWXlRa015Tnprd1JrSkVRemRCTWpBM1FqYzVOZyJ9.eyJlbWFpbCI6ImFsYmVydEBiZWtzdGlsLm5ldCIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJpc3MiOiJodHRwczovL2Rjb3MuYXV0aDAuY29tLyIsInN1YiI6Imdvb2dsZS1vYXV0aDJ8MTA5OTY0NDk5MDExMTA4OTA1MDUwIiwiYXVkIjoiM3lGNVRPU3pkbEk0NVExeHNweHplb0dCZTlmTnhtOW0iLCJleHAiOjIwOTA4ODQ5NzQsImlhdCI6MTQ2MDE2NDk3NH0.OxcoJJp06L1z2_41_p65FriEGkPzwFB_0pA9ULCvwvzJ8pJXw9hLbmsx-23aY2f-ydwJ7LSibL9i5NbQSR2riJWTcW4N7tLLCCMeFXKEK4hErN2hyxz71Fl765EjQSO5KD1A-HsOPr3ZZPoGTBjE0-EFtmXkSlHb1T2zd0Z8T5Z2-q96WkFoT6PiEdbrDA-e47LKtRmqsddnPZnp0xmMQdTr2MjpVgvqG7TlRvxDcYc-62rkwQXDNSWsW61FcKfQ-TRIZSf2GS9F9esDF4b5tRtrXcBNaorYa9ql0XAWH5W_ct4ylRNl3vwkYKWa4cmPvOqT5Wlj9Tf0af4lNO40PQ'
+__CLI_LOGIN_EE_USERNAME = 'bootstrapuser'
+__CLI_LOGIN_EE_PASSWORD = 'deleteme'
+
+log = logging.getLogger(__name__)
+
+
+def get_acs_token(dcosurl: str, username: str, password: str, is_enterprise: bool) -> None:
+    if is_enterprise:
+        log.info('logging into {} as {}'.format(dcosurl, username))
+        payload = {'uid': username, 'password': password}
+    else:
+        log.info('logging into {} with default open token'.format(dcosurl))
+        payload = {'token': __CLI_LOGIN_OPEN_TOKEN}
+
+    headers = {'Content-Type': 'application/json',}
+    login_endpoint = '{dcosurl}/acs/api/v1/auth/login'.format(dcosurl=dcosurl)
+    r = requests.post(login_endpoint, headers=headers, json=payload, verify=False)
+    assert r.status_code == 200, '{} failed {}: {}'.format(login_endpoint, r.status_code, r.text)
+
+    return r.json()['token']
+
+
+def login(dcosurl: str, username: str, password: str, is_enterprise: bool, user_token: str) -> str:
+    if not user_token:
+        user_token = get_acs_token(dcosurl, username, password, is_enterprise)
+    return user_token
+
+
+def configure_cli(dcosurl: str, token: str) -> None:
+    out, err, rc = shakedown.run_dcos_command('dcos config set core.dcos_url {}'.format(dcosurl))
+    assert rc, 'Failed to set core.dcos_url: {}'.format(err)
+    out, err, rc = shakedown.run_dcos_command('dcos config set core.ssl_verify false')
+    assert rc, 'Failed to set core.ssl_verify: {}'.format(err)
+    out, err, rc = shakedown.run_dcos_command('config set core.dcos_acs_token {}'.format(token))
+    assert rc, 'Failed to set core.dcos_acs_token: {}'.format(err)
```

Comment:
That's a fair point and I do mind just pushing that out to another issue where we apply it consistently. I definitely have a tendency to push for "landing" paradigms over maintaining consistency...

Given that we can agree to table this, my qualm with shakedown in this case is the accessibility of the interface. Once I use shakedown for a CLI command, I am extremely limited in what I can  do with that command: https://github.com/dcos/shakedown/blob/master/shakedown/dcos/command.py#L103

By using subprocess, I have the ability to:
* declare a timeout
* arbitrarily redirect stdout, stdin, and stderr
* implicitly raise an expception on failure
* specifically control the logging statements around this command
* specify if the command should be shell interpreted
* control the specific environment for that command
* provide my command in an unambiguous argument list

By using shakedown:
* I MUST submit my command as a string that will be split up again anyway. This makes it harder to provide intricate arguments.
* I MUST push all output to either subprocess.PIPE or stdout. This prevents me from streaming the output
* I MUST declare with a keyword arg that failures in the command should raise a local exception
* If I want to access any of the powerful subprocess options ( https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module ) then I need to make a commit to shakedown and then wait for a dependency update in this repo. This means I cannot control: timeouts, working dir, command entrypoint (shell), or custom decoders for I/O (might be useful for mesos task log retrieval)

TL;DR I see the argument on maintaining consistency so we can handle this later, but I really care about making sure tests can directly access the powerful, native subprocess interface instead of using a customized, restrictive interface